File: Frontend/src/services/cropService.ts
import api from './api';
import type { CropRequest, CropResponse } from '../types';

export const cropService = {
  listByFarmer: (farmerId: number) =>
    api.get<CropResponse[]>('/crops', { params: { farmerId } }),
  getById: (id: number) => api.get<CropResponse>(`/crops/${id}`),
  create: (data: CropRequest) => api.post<CropResponse>('/crops', data),
  update: (id: number, data: CropRequest) => api.put<CropResponse>(`/crops/${id}`, data),
  delete: (id: number) => api.delete(`/crops/${id}`),
};

File: Backend/src/main/java/com/smartagriculture/service/CropService.java
package com.smartagriculture.service;

import com.smartagriculture.dto.crop.CropRequest;
import com.smartagriculture.dto.crop.CropResponse;
import java.util.List;

public interface CropService {

    CropResponse create(CropRequest request);

    CropResponse update(Long id, CropRequest request);

    void delete(Long id);

    CropResponse getById(Long id);

    List<CropResponse> findByFarmer(Long farmerId);
}
